src/hooks/feature-cards/usePersonalizedFeatureCards.tsx

```diff
@@ -0,0 +1,53 @@
+import { useAccount } from '@lifi/wallet-management';
+import { useMemo } from 'react';
+import { useFeatureCardsFilter } from 'src/hooks/feature-cards/useFeatureCardsFilter';
+import { usePersonalizedFeatureCardsQuery } from 'src/hooks/feature-cards/usePersonalizedFeatureCardsQuery';
+import { usePersonalizedFeatureOnLevel } from 'src/hooks/feature-cards/usePersonalizedFeatureOnLevel';
+import { useLoyaltyPass } from 'src/hooks/useLoyaltyPass';
+import { useSettingsStore } from 'src/stores/settings';
+import { shallow } from 'zustand/shallow';
+
+export const usePersonalizedFeatureCards = () => {
+  const { account } = useAccount();
+  const { points } = useLoyaltyPass(account?.address);
+  const { excludedFeatureCardsFilter } = useFeatureCardsFilter();
+  const disabledFeatureCards = useSettingsStore(
+    (state) => state.disabledFeatureCards,
+    shallow,
+  );
+  const { data: featureCardsToDisplay } = usePersonalizedFeatureCardsQuery();
+  const { featureCards: featureCardsLevel } = usePersonalizedFeatureOnLevel({
+    points: points,
+    enabled: !!points,
+  });
+
+  const slicedPersonalizedFeatureCards = useMemo(() => {
+    const personalizedFeatureCards =
+      featureCardsToDisplay && featureCardsToDisplay.length > 0
+        ? featureCardsToDisplay
+        : featureCardsLevel && featureCardsLevel.length > 0
+          ? [featureCardsLevel[0]]
+          : undefined;
+
+    if (
+      Array.isArray(personalizedFeatureCards) &&
+      !!personalizedFeatureCards.length
+    ) {
+      return personalizedFeatureCards
```

Comment:
duplication of https://github.com/jumperexchange/jumper-exchange/pull/1659/files#diff-0ca5aad46cf969d7ae6a1b9d9fe2add1298cd7d3c2012812df66dd00937b10f1R26